webpack/assets/javascripts/react_app/components/common/forms/OrderableSelect/OrderableSelect.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { set } from 'lodash';
+import { TypeAheadSelect } from 'patternfly-react';
+
+import { noop } from '../../../../common/helpers';
+import { orderDragged } from './helpers';
+import OrderableToken from './components/OrderableToken';
+
+/**
+ * Wraps TypeAheadSelect with an Orderable HOC.
+ * Presumes to be wrapped in a DndProvider context.
+ */
+const OrderableSelect = ({
+  onChange,
+  defaultValue,
+  value,
+  options,
+  ...props
+}) => {
+  const [internalValue, setInternalValue] = useState(
```

Comment:
does it make sense to keep the value as an internal data? other components need to access it? if so, then the state should be global via redux

Reply:
We keep it as an internal state, because we are not touching external value and we need to keep the order of the value somewhere. And no, other components should not need it directly, they should use `onChange` prop. We currently use `value` and `defaultValue` only as defaultValues, we can't change value through props, that's not needed in common cases though.